src/pages/SearchPage.js

```diff
@@ -1,10 +1,126 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
+import SearchCard from "../components/SearchPageComponents/SearchCard/SearchCard";
+import { Stack, Typography, Button } from "@mui/material";
+import SidePanel from "../components/SearchPageComponents/SidePanel/SidePanel";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import axios from "axios";
+import { useParams } from "react-router-dom";
 
 const SearchPage = () => {
+  const [list, setList] = useState([]);
+  const [resultsNum, setResultsNum] = useState(5);
+  const { searchText } = useParams();
+
+  useEffect(() => {
```

Comment:
initiate the options object and the fetchData outside the useEffect

Reply:
done